test/node-test/client-connect.js

```diff
@@ -24,97 +25,101 @@ test('basic connect', (t) => {
       socket.end(data)
     })
   })
-  t.teardown(server.close.bind(server))
+  t.after(server.close.bind(server))
 
   server.listen(0, async () => {
     const client = new Client(`http://localhost:${server.address().port}`)
-    t.teardown(client.close.bind(client))
+    t.after(() => client.close.bind(client)())
```

Comment:
It seems the function is getting called instead of passed

Reply:
I think the two codes below are equivalent:

```js
t.after(() => client.close.bind(client)())
```

```js
t.after(() => {
  return client.close.bind(client)()
})
```

However, to make it easier to understand, I rewritten it as `t.after(() => { client.close() })`

[e44332a](https://github.com/nodejs/undici/pull/2591/commits/e44332a81d42201c224edc83f0c54f9fb6b47c88)